org.civicrm.reqangular/src/common/services/api/option-group.js

```diff
@@ -1,95 +1,82 @@
+     *
+     * @param {array} optionValues
+     * @return {object} an object formed as such:
+     *  {
+     *    group_name_1: [ { value_1 }, { ... }, { value_n } ],
+     *    ...
+     *    group_name_n: [ { value_1 }, { ... }, { value_n } ]
+     *  }
+     */
+    function valuesByGroup (optionValues) {
+      return _.transform(optionValues, function (result, optionValue) {
+        var optionGroup = optionValue.option_group_name;
 
-                return idsByName;
-            });
-        }
+        result[optionGroup] = result[optionGroup] || [];
+        result[optionGroup].push(optionValue);
+      });
+    }
 
-        return api.extend({
+    return api.extend({
 
-            /**
-             * Returns the values of the option groups with the given names
-             *
-             * @param {string/array} groupNames
-             *   If the value is an array of names, the method will group
-             *   the values by option group names
-             * @param {object} optional parameters for the query
-             * @return {Promise}
-             *   Resolves to an array with the values (if `groupNames` is a string)
-             *   or an object with keys as the group names and values as the
-             *   array of their option values (if `groupNames` is an array)
-             */
-            valuesOf: function (groupNames, params) {
-                var multiple = _.isArray(groupNames);
+      /**
+       * Returns the values of the option groups with the given names
+       *
+       * @param {string/array} groupNames
+       *   If the value is an array of names, the method will group
+       *   the values by option group names
+       * @param {object} optional parameters for the query
+       * @return {Promise}
+       *   Resolves to an array with the values (if `groupNames` is a string)
+       *   or an object with keys as the group names and values as the
+       *   array of their option values (if `groupNames` is an array)
+       */
+      valuesOf: function (groupNames, params) {
+        var multiple = _.isArray(groupNames);
 
-                return idsOf.call(this, multiple ? groupNames : [groupNames])
-                    .then(function (groupIds) {
-                        return this.sendGET('OptionValue', 'get', angular.extend({
-                            option_group_id: { 'IN': _.values(groupIds) },
-                            is_active: '1'
-                        }, params))
-                        .then(function (data) {
-                            if (multiple) {
-                                return valuesByGroup(data.values, groupIds);
-                            } else {
-                                return data.values;
-                            }
-                        });
-                    }.bind(this));
-            }
+        return this.sendGET('OptionValue', 'get', _.assign({
+          'option_group_id.name': { 'IN': multiple ? groupNames : [groupNames] },
```

Comment:
You can check yourself, but when using the PHP API this explicit join isn't required, `option_group_id` accepts _either_ integers for the group IDs or strings for the group names, example from API explorer:

```
CRM.api3('OptionValue', 'get', {
  "option_group_id.name": {"IN":["account_relationship","activity_contacts"]}
}).done(function(result) {
  // do something
});
```